Build fails to configure with gcc 4.9.2

```
...
[libint] checking gmpxx.h usability... yes
[libint] checking gmpxx.h presence... yes
[libint] checking for gmpxx.h... yes
[libint] checking if GMP C++ library is usable... no
[libint] configure: error: cannot link against GMP C++ library. Please get the latest GMP with support for C++. It can be obtained at gmplib.org .
```
The `config.log` says:
```
configure:6514: g++ -o conftest  -O2  -std=c++11 -I/local/certik/bld/autoconf/syuobdyszfxn/include -I/local/certik/bld/automake/jjtcusj34hdq/include -I/local/certik/bld/gmp/vzmznaezaap7/include -I/local/certik/bld/patchelf/eokqj5ouftcf/include -L/local/certik/bld/autoconf/syuobdyszfxn/lib -Wl,-rpath=/local/certik/bld/autoconf/syuobdyszfxn/lib -L/local/certik/bld/automake/jjtcusj34hdq/lib -Wl,-rpath=/local/certik/bld/automake/jjtcusj34hdq/lib -L/local/certik/bld/gmp/vzmznaezaap7/lib -Wl,-rpath=/local/certik/bld/gmp/vzmznaezaap7/lib -L/local/certik/bld/patchelf/eokqj5ouftcf/lib -Wl,-rpath=/local/certik/bld/patchelf/eokqj5ouftcf/lib conftest.cpp -lgmpxx -lgmp  >&5
In file included from /local/certik/bld/gmp/vzmznaezaap7/include/gmp.h:51:0,
                 from /local/certik/bld/gmp/vzmznaezaap7/include/gmpxx.h:33,
                 from conftest.cpp:57:
/local/certik/bld/gcc/vfnflgljyg4l/include/c++/4.9.2/cstddef:51:11: error: '::max_align_t' has not been declared
   using ::max_align_t;
           ^
```
That's a well known error (https://gcc.gnu.org/gcc-4.9/porting_to.html) and the solution is to put `<cstddef>` before `<gmpxx.h>`.

--- a/lib/autoconf/acx_check_gmpxx.m4
+++ b/lib/autoconf/acx_check_gmpxx.m4
@@ -11,7 +11,8 @@ AC_DEFUN([ACX_CHECK_GMPXX], [
   LIBS="-lgmpxx -lgmp $LIBS"
   AC_LINK_IFELSE(
    [AC_LANG_PROGRAM(
-    [[#include <gmpxx.h>
+    [[#include <cstddef>
+      #include <gmpxx.h>
     ]],
     [[
      mpf_class a = 1.0;

--- a/src/bin/libint/key.h
+++ b/src/bin/libint/key.h
@@ -21,6 +21,7 @@
 #define _libint2_src_bin_libint_key_h_
 
 #include <libint2/intrinsic_types.h>
+#include <cstddef>
 #include <gmpxx.h>
 #include <sstream>
 

--- a/src/bin/test_eri/eri.h
+++ b/src/bin/test_eri/eri.h
@@ -24,6 +24,7 @@
 
 #if defined(__cplusplus)
 #if HAVE_MPFR
+# include <cstddef>
 # include <gmpxx.h>
 # include <mpfr.h>
  typedef mpf_class LIBINT2_REF_REALTYPE;

